### PR TITLE
Plug-ins for extending lizzie without edit source

### PR DESCRIPTION
**Pull request by cngoodboy**

### CODE REVIEW

File: src/wagner/stephanie/lizzie/gui/BoardRenderer.java [hunk 1]
```diff
@@ -21,7 +22,7 @@
 import java.util.Map;
 
 public class BoardRenderer {
-    private static final double MARGIN = 0.03; // percentage of the boardLength to offset before drawing black lines
+    private static final double MARGIN = 0.05; // percentage of the boardLength to offset before drawing black lines
```
Comment (featurecat) [inline src/wagner/stephanie/lizzie/gui/BoardRenderer.java]: why was this changed? maybe we can let it be customizable though.

File: src/wagner/stephanie/lizzie/gui/BoardRenderer.java [hunk 2]
```diff
@@ -199,10 +204,6 @@ private void drawBranch() {
         bestMoves = Lizzie.leelaz.getBestMoves();
         branch = null;
 
-        // We can't early-out until now, since we need bestMoves for later
-        if (!Lizzie.config.showVariation)
```
[the remark this answers is not in the document]
Comment (cngoodboy) [reply src/wagner/stephanie/lizzie/gui/BoardRenderer.java]: Here his function and my function are duplicated.

File: src/wagner/stephanie/lizzie/gui/BoardRenderer.java (hunk 1 above)
Comment (cngoodboy) [reply src/wagner/stephanie/lizzie/gui/BoardRenderer.java]: Aerisnju 's suggest

File: src/wagner/stephanie/lizzie/gui/BoardRenderer.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (dfannius) [reply src/wagner/stephanie/lizzie/gui/BoardRenderer.java]: I can’t test your code right now, but I want to make sure that the variation code still works. I had to put my test inside `drawBranch` because `getBestMoves` is called inside `drawBranch` (not ideal, in my opinion, but I tried to be minimally invasive) which we still need to happen so that the draw-suggestions code can access `bestMoves`. It’s not obvious to me from glancing at this code that this still works - can you check, and fix if not? Just toggle variations with `v` and see if suggestions are still displayed.

[the remark this answers is not in the document]
Comment (featurecat) [reply src/wagner/stephanie/lizzie/gui/BoardRenderer.java]: I already merged dfannius's code, so I'd prefer to keep whats existing already if it isn't different.

[the remark this answers is not in the document]
Comment (cngoodboy) [reply src/wagner/stephanie/lizzie/gui/BoardRenderer.java]: OK

**Comment by cngoodboy** [comment]
Inadvertently added the contents of another Pull Request to the master branch
